pkg/controllers/vrpq/vdbverify_reconciler.go

```diff
@@ -0,0 +1,92 @@
+	v1beta1 "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	vrpqstatus "github.com/vertica/vertica-kubernetes/pkg/vrpqstatus"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const stateIncompatibleDB = "Incompatible"
+
+type VDBVerifyReconciler struct {
+	VRec *VerticaRestorePointsQueryReconciler
+	Vrpq *v1beta1.VerticaRestorePointsQuery
+	Log  logr.Logger
+}
+
+func MakeVDBVerifyReconciler(r *VerticaRestorePointsQueryReconciler, vrpq *v1beta1.VerticaRestorePointsQuery,
+	log logr.Logger) controllers.ReconcileActor {
+	return &VDBVerifyReconciler{
+		VRec: r,
+		Vrpq: vrpq,
+		Log:  log.WithName("VDBVerifyReconciler"),
+	}
+}
+
+// Reconcile will verify the VerticaDB in the Vrpq CR exists, vclusterops is enabled and
+// the vertica version supports vclusterops deployment
+func (q *VDBVerifyReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	vdb := &vapi.VerticaDB{}
+	nm := names.GenNamespacedName(q.Vrpq, q.Vrpq.Spec.VerticaDBName)
+	if res, err := vk8s.FetchVDB(ctx, q.VRec, q.Vrpq, nm, vdb); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+
+	// check version for Vdb, the minimim version should be 24.2.0
+	vinf, err := vdb.MakeVersionInfoCheck()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !vinf.IsEqualOrNewer(vapi.RestoreSupportedMinVersion) {
+		q.VRec.Eventf(q.Vrpq, corev1.EventTypeWarning, events.RestoreNotSupported,
+			"The Vertica version '%s' doesn't support in-database restore points", vinf.VdbVer)
+		err = vrpqstatus.Update(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+			[]*metav1.Condition{vapi.MakeCondition(v1beta1.QueryReady, metav1.ConditionFalse, "IncompatibleDB")}, stateIncompatibleDB, nil)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Should be deployed with vclusterops, not supported for admintools deployments
+	if !vmeta.UseVClusterOps(vdb.Annotations) {
+		q.VRec.Event(q.Vrpq, corev1.EventTypeWarning, events.AdmintoolsNotSupported,
+			"ShowRestorePoints is not supported for admintools deployments")
+		err = vrpqstatus.Update(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+			[]*metav1.Condition{vapi.MakeCondition(v1beta1.QueryReady, metav1.ConditionFalse, "AdmintoolsNotSupported")}, stateIncompatibleDB, nil)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+	return ctrl.Result{}, vrpqstatus.Update(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		[]*metav1.Condition{vapi.MakeCondition(v1beta1.QueryReady, metav1.ConditionTrue, "Started")}, stateQuerying, nil)
```

Comment:
The reason for the QueryReady status condition is done at the point. So suggestion this:
```suggestion
		[]*metav1.Condition{vapi.MakeCondition(v1beta1.QueryReady, metav1.ConditionTrue, "Complete")}, stateQuerying, nil)
```